source/common/http/utility.h

```diff
@@ -668,6 +669,29 @@ bool isSafeRequest(const Http::RequestHeaderMap& request_headers);
  */
 Http::Code maybeRequestTimeoutCode(bool remote_decode_complete);
 
+/**
+ * Container for route config elements that pertain to a redirect.
+ */
+struct RedirectConfig {
+  const std::string scheme_redirect_;
+  const std::string host_redirect_;
+  const std::string port_redirect_;
+  const std::string path_redirect_;
+  const std::string prefix_rewrite_redirect_;
+  std::string regex_rewrite_redirect_substitution_;
```

Comment:
I think you can make these fields `const` as well. You can use ASSERT after the structure was created to check for mutually exclusive fields.